src/compiler/checker/lines_4000_8000.rs

```diff
@@ -318,15 +317,11 @@ pub(super) fn create_node_builder() -> NodeBuilder {
 }
 
 #[derive(Debug)]
-pub struct NodeBuilder {
-    synthetic_factory: BaseNodeFactorySynthetic,
```

Comment:
Updated the synthetic base node factory to be a `thread_local!` so updated to inject that value by calling `.with()` on the `thread_local!` at each point of usage